cs-jClouds/src/main/java/io/cloudslang/content/jclouds/services/helpers/AWSSignatureHelper.java

```diff
@@ -0,0 +1,150 @@
+
+        StringBuilder queryString = new StringBuilder();
+        for (Map.Entry<String, String> entry : sortedList) {
+            queryString.append(entryToQuery(entry));
+        }
+        if (queryString.length() > 0) queryString.deleteCharAt(queryString.length() - 1);   //last &
+
+        return queryString.toString();
+    }
+
+    /**
+     * The canonicalized (standardized) headers string is formed by first sorting all the header
+     * parameters, then converting all header names to lowercase and
+     * trimming excess white space characters out of the header values
+     *
+     * @param headers The headers to be canonicalized.
+     * @return A canonicalized form for the specified headers.
+     */
+    public String canonicalizedHeadersString(Map<String, String> headers) {
+        List<Map.Entry<String, String>> sortedList = new ArrayList<>(headers.entrySet());
+        Collections.sort(sortedList, new Comparator<Map.Entry<String, String>>() {
+            public int compare(Map.Entry<String, String> o1, Map.Entry<String, String> o2) {
+                return (o1.getKey().compareToIgnoreCase(o2.getKey()));
+            }
+        });
+
+        String header;
+        String headerValue;
+        StringBuilder headerString = new StringBuilder();
+
+        for (Map.Entry<String, String> ent : sortedList) {
+            header = nullToEmpty(ent.getKey()).toLowerCase();
+            headerValue = nullToEmpty(ent.getValue()).trim();
+            headerString.append(header).append(Constants.Miscellaneous.COLON)
+                    .append(headerValue).append(Constants.Miscellaneous.LINE_SEPARATOR);
+        }
+
+        return headerString.toString();
+    }
+
+    /**
+     * Creates a comma separated list of headers.
+     *
+     * @param headers The headers to be signed.
+     * @return The comma separated list of headers to be signed.
+     */
+    public String getSignedHeadersString(Map<String, String> headers) {
+        List<String> sortedList = new ArrayList<>(headers.keySet());
+        Collections.sort(sortedList, String.CASE_INSENSITIVE_ORDER);
+
+        StringBuilder signedHeaderString = new StringBuilder();
+        for (String header : sortedList) {
+            if (signedHeaderString.length() > 0)
+                signedHeaderString.append(SEMICOLON);
+            signedHeaderString.append(nullToEmpty(header).toLowerCase());
+        }
+        return signedHeaderString.toString();
+    }
+
+    /**
+     * Converts a java.util.Date to an AWS specific date format.
+     * The AWS date should be in (java) format "yyyyMMdd'T'HHmmss'Z'" and in time zone UTC.
+     *
+     * @param date The date to be formatted.
+     * @return A string representing the formatted string.
+     */
+    public String getAmazonDateString(Date date) {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_FORMAT);
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
```

Comment:
@gituser173 Yes, because there are different Amazon regions, on different time shift... so I hard-coded the UTC to have a certain reference point.